ml-agents-envs/mlagents/envs/tests/test_sampler_class.py

```diff
@@ -0,0 +1,154 @@
+from math import isclose
+import pytest
+
+from mlagents.envs.sampler_class import SamplerManager
+from mlagents.envs.sampler_class import UniformSampler, MultiRangeUniformSampler
+from mlagents.envs.exception import UnityException
+
+
+def basic_3Dball_sampler():
+    return {
+        "mass":
+            {
+                "sampler-type": "uniform",
+                "min_value": 5,
+                "max_value": 10
+            },
+        "gravity":
+            {
+                "sampler-type": "multirange_uniform",
+                "intervals": [[8, 11], [15, 20]]
+            }
+    }
+
+
+def check_value_in_intervals(val, intervals):
+    check_in_bounds = [a <= val <= b for a, b in intervals]
+    return any(check_in_bounds)
+
+def test_3Dball_sampler():
+    config = basic_3Dball_sampler()
+    sampler = SamplerManager(config)
+
+    assert sampler.check_empty_sampler_manager() == False
+    assert isinstance(sampler.samplers["mass"], UniformSampler)
+    assert isinstance(sampler.samplers["gravity"], MultiRangeUniformSampler)
+
+    cur_sample = sampler.sample_all()
+
+    # Check uniform sampler for mass
+    assert sampler.samplers["mass"].min_value == config["mass"]["min_value"]
+    assert sampler.samplers["mass"].max_value == config["mass"]["max_value"]
+    assert config["mass"]["min_value"] <= cur_sample["mass"]
+    assert config["mass"]["max_value"] >= cur_sample["mass"]
+
+    # Check multirange_uniform sampler for gravity
+    assert sampler.samplers["gravity"].intervals == config["gravity"]["intervals"]
+    assert check_value_in_intervals(cur_sample["gravity"], sampler.samplers["gravity"].intervals)
+
+def basic_tennis_sampler():
+    return {
+        "angle":
```

Comment:
Rather than configs specific to an environment, these tests should be covering the code that you wrote. AFAICT this test checks the same thing as the previous one (uniform and multirange_uniform samplers) and neither checks the Gaussian one.